Document the chef_license config option (#1785)
test-kitchen/test-kitchen: docs/content/docs/provisioners/chef.md

    - chef-23 # deprecation numeric ID
    - recipes/install.rb:22 # specific line in a file
  product_name: chef # which package to install chef || chef-workstation

>   chef_license: nil # to accept the Chef Infra license automatically in Test Kitchen. Accepts `accept`, `accept-no-persist` and `accept-silent`

Reviewer: ```suggestion
  chef_license: nil # to accept the Chef Infra license automatically in Test Kitchen. Valid values: `accept`, `accept-no-persist`, or `accept-silent`
```
